pennylane/transforms/decompositions/single_qubit_unitary.py

```diff
@@ -112,3 +112,59 @@ def zyz_decomposition(U, wire):
     phis, thetas, omegas = map(math.squeeze, [phis, thetas, omegas])
 
     return [qml.Rot(phis, thetas, omegas, wires=wire)]
+
+
+def xyx_decomposition(U, wire):
+    r"""Recover the decomposition of a single-qubit matrix :math:`U` in terms
+    of elementary operations, as a product of X and Y rotations in the form
+    :math:`e^{i\gamma} RX(\phi) RY(\theta) RX(\lambda)`.
+
+    Args:
+        U (tensor): A 2 x 2 unitary matrix.
+        wire (Union[Wires, Sequence[int] or int]): The wire on which to apply the operation.
+
+    Returns:
+        list[qml.Operation]: Returns a list of 4 gates, an ``Identity`` times
+        the global phase, then an ``RX``, an ``RY``and another ``RX`` gate,
+        which when applied in the order of appearance in the list is equivalent
+        to the unitary :math:`U`.
+
+    **Example**
+
+    >>> U = np.array([[-0.28829348-0.78829734j,  0.30364367+0.45085995j],
+                      [ 0.53396245-0.10177564j,  0.76279558-0.35024096j]])
+    >>> decomp = xyx_decomposition(U, 0)
+    >>> decomp
+    [(0.38469215914523336-0.9230449299422961j)*(Identity(wires=[0])),
+            RX(0.45246583660683803, wires=[0]),
+            RY(1.3974974118006183, wires=[0]),
+            RX(-1.7210192479534632, wires=[0])]
+    """
+
+    # Small number to add to denominators to avoid division by zero
+    EPS = 1e-64
+
+    # Choose gamma such that exp(-i*gamma)*U is special unitary (detU==1).
+    gamma = math.cast_like(math.angle(math.linalg.det(U)) / 2, 1j)
+    U_det1 = math.exp(-1j * gamma) * U
+
+    # Compute \phi, \theta and \lambda after analytically solving for them from
+    # U_det1 = expm(1j*\phi*PauliX) expm(1j*\theta*PauliY) expm(1j*\lambda*PauliX)
+    lam_plus_phi = math.arctan2(-math.imag(U_det1[0, 1]), math.real(U_det1[0, 0]) + EPS)
+    lam_minus_phi = math.arctan2(math.imag(U_det1[0, 0]), -math.real(U_det1[0, 1]) + EPS)
+    lam = lam_plus_phi + lam_minus_phi
+    phi = lam_plus_phi - lam_minus_phi
+
+    # The following conditional attempts to avoid 0 / 0 errors. Either the
+    # sine is 0 or the cosine, but not both.
+    if math.allclose(lam_plus_phi, 0):
+        theta = 2 * math.arccos(math.real(U_det1[1, 1]) / (math.cos(lam_plus_phi) + EPS))
+    else:
+        theta = 2 * math.arccos(-math.imag(U_det1[0, 1]) / (math.sin(lam_plus_phi) + EPS))
+
+    return [
+        qml.s_prod(math.exp(1j * gamma), qml.Identity(wire)),
```

Comment:
That's a great trick to add the globaphase! Could make it optional though? I think most hardware would not understand it. You could introduce a `global_phase` boolean arg and turn it off as default.

Reply:
I've added the flag `return_global_phase` (and also modified `_convert_to_su2` to accept it). Changed the order in the list s.t. the global phase operators is the last, if returned.